x-pack/plugins/cloud_security_posture/server/routes/benchmarks/v2.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ListResult, PackagePolicy } from '@kbn/fleet-plugin/common';
+import { transformError } from '@kbn/securitysolution-es-utils';
+import type { Logger } from '@kbn/core/server';
+import {
+  PackagePolicyClient,
+  AgentPolicyServiceInterface,
+  AgentService,
+} from '@kbn/fleet-plugin/server';
+import { SavedObjectsClientContract } from '@kbn/core-saved-objects-api-server';
+import { CLOUD_SECURITY_POSTURE_PACKAGE_NAME, POSTURE_TYPE_ALL } from '../../../common/constants';
+import {
+  getCspPackagePolicies,
+  getCspAgentPolicies,
+  getAgentStatusesByAgentPolicies,
+} from '../../lib/fleet_util';
+import { CspApiRequestHandlerContext } from '../../types';
+import { createBenchmarks, getBenchmarks } from './utilities';
+
+export const getBenchmark = async (
+  // context: CspRequestHandlerContext,
+  // request: any,
+  cspContext: CspApiRequestHandlerContext,
+  soClient: SavedObjectsClientContract,
+  packagePolicyService: PackagePolicyClient,
+  query: any,
+  agentPolicyService: AgentPolicyServiceInterface,
+  agentService: AgentService,
+  logger: Logger,
+  responseOk?: any,
+  responseError?: any
```

Comment:
we wish to separate between the facade layer and the BL layer such that the facade layer will interact with the BL layer, without the BL layer is coupled to the facade.

Here, passing response objects or `CspApiRequestHandlerContext` breaks that design